openupgrade_scripts/scripts/mail/15.0.1.5/post-migration.py

```diff
@@ -39,6 +39,30 @@ def _map_mail_notification_failure_type(env):
     )
 
 
+def finish_migration_to_mail_group(env):
+    env.cr.execute(
+        """
+        SELECT 1 FROM ir_module_module
+        WHERE name = 'mail_group' AND state = 'to install'""",
+    )
+    will_have_mail_group = env.cr.rowcount
+    if not will_have_mail_group:
+        return
+    openupgrade.logged_query(
+        env.cr,
+        """
+        SELECT mc.id
+        FROM mail_channel mc
+        WHERE mc.email_send""",
+    )
+    old_mail_groups = env["mail.channel"].browse([x[0] for x in env.cr.fetchall()])
+    if old_mail_groups:
+        try:
+            old_mail_groups.unlink()
```

Comment:
Why doing this in ORM?

Reply:
And shouldn't you remove one by one?

Reply:
Do you think it's better to delete the records by query?

Reply:
We will avoid side effects

Reply:
Yeah, I saw those effects, due to cascade deletions, and produced issues. Better by query, and thus no issues :)